### PR TITLE
Add solver_microservice cookbook with setup recipe

### PR DESCRIPTION
**Pull request by arturtr**
It installs pipenv, sync assignment repo and starts microservice

https://dekeo.atlassian.net/browse/ALPHA-4292

### CODE REVIEW

File: solver_microservice/recipes/sync.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  user params[:deploy][:user]
+  group params[:deploy][:group]
+
+  repository 'git@github.com:sdtechdev/assignment.git'
+  revision 'main'
+  action :sync
+end
+
+bash 'solver_start' do
+  user params[:deploy][:user]
```
Comment (shkrt) [inline solver_microservice/recipes/sync.rb]: `user node['deployer']['user'] `

File: solver_microservice/recipes/sync.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  cwd SOLVER_DIR
```
Comment (shkrt) [inline solver_microservice/recipes/sync.rb]: let's instead use node attributes, i.e `node['assignment_solver']['working_dir']` and get rid of this constant

File: solver_microservice/recipes/sync.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+pip 'pipenv'
+
+SOLVER_DIR = '/srv/www/solver'
+
+git SOLVER_DIR do
+  user params[:deploy][:user]
+  group params[:deploy][:group]
+
+  repository 'git@github.com:sdtechdev/assignment.git'
```
Comment (shkrt) [inline solver_microservice/recipes/sync.rb]: let's use node attributes here, `node['assignment_solver']['repo_url']`

File: solver_microservice/recipes/sync.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+pip 'pipenv'
+
+SOLVER_DIR = '/srv/www/solver'
+
+git SOLVER_DIR do
+  user params[:deploy][:user]
+  group params[:deploy][:group]
```
Comment (shkrt) [inline solver_microservice/recipes/sync.rb]:
```
user node['deployer']['user']
group node['deployer']['group']
```

File: solver_microservice/recipes/sync.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  action :sync
+end
+
+bash 'solver_start' do
+  user params[:deploy][:user]
+  cwd SOLVER_DIR
+
+  code <<~SH
+    pipenv install
+    pipenv run gunicorn -b 127.0.0.1:5000 -D application:application
```
Comment (shkrt) [inline solver_microservice/recipes/sync.rb]: host and port also could be extracted to node attributes

**Review by shkrt — COMMENTED** [review]
Let's extract everything as suggested to node attributes. Node attributes can be specified in layer settings or stack settings (if needs to run on multiple layers). So each time we want to change these, we only change stack settings without need to update a repo. Plus this is still a public repo and we want to make this as generic as possible. 
Another question - at what lifecyle hook do we want to run this? If it will run during deploy, will it start new instance of gunicorn on each deploy? If so, then we only want to run it during setup hook.

**Comment by ashishtajane** [comment]
closing for now

**Pull request closed without merging**